### DISCUSSION TITLE
Installing extensions via command line is hanging or extremely slow.

### DISCUSSION THREAD

**Discussion by Dimss** [coder/code-server#2102] [Q&A]
I'm running `code-server` on OpenShift 3.11 cluster behind http proxy.  
I'm able to run `code-server` without any issue and access to the UI through the web browser. 
But installing simple extension through the cli (the  extension `.vsix` file downloaded and imported manually), for example ` ./code-server --disable-telemetry --install-extension ./gitlens.vsix` taking unreasonable long time, and can take up to 1h or even more. 
When I'm running exactly the same command in my local mac, everything works just fine (also when I disconnect my mac from internet). 
I don't know how to proceed with this issue, I'm trying to run the install extension command with `--verbose` flag, but logs messages are stoped at `Installing extension...`, and not clear what is going on.
Any help will be appreciated. 
![Screen Shot 2020-09-16 at 18 24 08](https://user-images.githubusercontent.com/5563811/93360182-0d25b780-f84c-11ea-99b6-67705aac9b5c.png)

**Comment by code-asher** [comment]
Do you experience similar slowness with plain `curl`? For example to test querying our API:

```
$ curl 'https://extensions.coder.com/api/extensionquery' --data-raw '{}'
{"results":[]}
```

Or to download an extension manifest:
```
$ https://storage.googleapis.com/vscode-extension-assets/vscode-extensions/vscode/eamodio/gitlens/10.2.2/Microsoft.VisualStudio.Code.Manifest
{
  "name":  "gitlens",
  ...
}
```

**Reply by Dimss** [reply]
yes, indeed, the request just hang. 
but when I'm executing from my local mac, it's work jus fine. 
but, one note, I'm installing extension locally, not over the network, is that still require network access?

**Reply by Dimss** [reply]
basically it's a kind of air gap env, the environment has access to limited internet urls, this is why I'm trying to install the extension locally with the `gitlens.vsix` file.

**Reply by code-asher** [reply]
Ohhhh my bad somehow I missed the part where you were installing a local vsix.

I think that for some reason VS Code still makes a request even when installing an extension locally. We've had a couple other reports about this but I haven't done any digging yet to see what's going on.

We should definitely fix it so that if you install a local extension it doesn't try to make any external requests.

**Reply by code-asher** [reply]
Opened an issue to fix this: https://github.com/cdr/code-server/issues/2103

**Reply by echuber2** [reply]
I'm interested in this use case too. Does it help if you add this in your user settings?

```json
"telemetry.enableTelemetry": false
```

If it works after an hour, does this mean it's just timing out with the telemetry and continuing anyway? Then would one workaround be to deactivate network interfaces temporarily for the install, or to filter the telemetry domains in the hosts file?

**Comment by Dimss** [comment]
Well, I tested it with --disable-telemetry flag, I think I tested it both
with that flag + setting  "telemetry.enableTelemetry": false in
config.yaml. But still got the same effect.
What is interesting, is when I do the same installation test on a computer
that is completely disconnected from the internet, the install does work.
I think the code server tries to connect to some external sources, but
maybe has some high value for timeout. So, in a proxy based environment,
where it does connected to the internet, but may block request on proxy/FW
drop request, code server still waiting for response due to high timeout
value, or sth like that.
